src/components/TypingIndicator/TypingIndicator.vue

```diff
@@ -0,0 +1,161 @@
+<!--
+  - @copyright Copyright (c) 2023 Maksim Sukharev <antreesy.web@gmail.com>
+  -
+  - @author Maksim Sukharev <antreesy.web@gmail.com>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<template>
+	<div v-show="typingParticipants.length" class="indicator">
+		<div class="indicator__wrapper">
+			<div class="indicator__avatars">
+				<AvatarWrapper v-for="(participant, index) in visibleParticipants"
+					:id="participant.actorId"
+					:key="index"
+					:name="participant.displayName"
+					:source="participant.actorType"
+					small
+					condensed
+					:condensed-overlap="8"
+					disable-menu
+					disable-tooltip />
+			</div>
+			<p class="indicator__main" v-html="indicatorMessage" />
+		</div>
+	</div>
+</template>
+
+<script>
+import escapeHtml from 'escape-html'
+
+import AvatarWrapper from '../AvatarWrapper/AvatarWrapper.vue'
+
+export default {
+	name: 'TypingIndicator',
+	components: { AvatarWrapper },
+
+	props: {
+		/**
+		 * The conversation token
+		 */
+		token: {
+			type: String,
+			required: true,
+		},
+	},
+
+	computed: {
+		participants() {
+			return this.$store.getters.participantsList(this.token).filter(participant => {
+				return participant.actorId !== this.$store.getters.getActorId()
+			})
+		},
+
+		typingParticipants() {
+			// TODO implement signal receiving here
+			return this.participants
+		},
+
+		visibleParticipants() {
+			return this.typingParticipants.slice(0, 3)
+		},
+
+		hiddenParticipantsCount() {
+			return this.typingParticipants.slice(3).length
+		},
+
+		indicatorMessage() {
+			if (!this.typingParticipants) {
+				return ''
+			}
+
+			const [user1, user2, user3] = this.prepareNamesList()
+
+			if (this.typingParticipants.length === 1) {
+				return t('spreed', '{user1} is typing...',
```

Comment:
Just a detail here @Antreesy – instead of 3 dots we should use the actual ellipsis character, and with a space between the text and it, like so: ` …`

> Maksim is typing …

Reply:
Thanks, did it in other PR, but forgot to change it here as well.
Will be shipped with the next commit, when API is merged!